diff --git a/src/Makefile b/src/Makefile
index 75a0bee..62899fb 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -13,7 +13,7 @@ RES       ?= res.bin
 
 EEPROM_SUPPORT = yes
 
-SRCS = flashcmd_api.c spi_controller.c spi_nand_flash.c spi_nor_flash.c ch341a_spi.c timer.c main.c
+SRCS = flashcmd_api.c ch341a_spi.c timer.c main.c
 
 ifdef LIBS_BASE
 CFLAGS += -I$(LIBS_BASE)/include
diff --git a/src/flashcmd_api.c b/src/flashcmd_api.c
index 746f1fc..3e67f0b 100644
--- a/src/flashcmd_api.c
+++ b/src/flashcmd_api.c
@@ -29,19 +29,7 @@ long flash_cmd_init(struct flash_cmd *cmd)
 	long flen = -1;
 
 #ifdef EEPROM_SUPPORT
-	if ((eepromsize <= 0) && (mw_eepromsize <= 0)) {
-#endif
-		if ((flen = snand_init()) > 0) {
-			cmd->flash_erase = snand_erase;
-			cmd->flash_write = snand_write;
-			cmd->flash_read  = snand_read;
-		} else if ((flen = snor_init()) > 0) {
-			cmd->flash_erase = snor_erase;
-			cmd->flash_write = snor_write;
-			cmd->flash_read  = snor_read;
-		}
-#ifdef EEPROM_SUPPORT
-	} else if ((eepromsize > 0) || (mw_eepromsize > 0)) {
+	if ((eepromsize > 0) || (mw_eepromsize > 0)) {
 		if ((eepromsize > 0) && (flen = i2c_init()) > 0) {
 			cmd->flash_erase = i2c_eeprom_erase;
 			cmd->flash_write = i2c_eeprom_write;
@@ -61,11 +49,7 @@ long flash_cmd_init(struct flash_cmd *cmd)
 
 void support_flash_list(void)
 {
-	support_snand_list();
-	printf("\n");
-	support_snor_list();
 #ifdef EEPROM_SUPPORT
-	printf("\n");
 	support_i2c_eeprom_list();
 	printf("\n");
 	support_mw_eeprom_list();
diff --git a/src/flashcmd_api.h b/src/flashcmd_api.h
index a50ad73..412c9d1 100644
--- a/src/flashcmd_api.h
+++ b/src/flashcmd_api.h
@@ -15,8 +15,6 @@
 #ifndef __FLASHCMD_API_H__
 #define __FLASHCMD_API_H__
 
-#include "nandcmd_api.h"
-#include "snorcmd_api.h"
 #ifdef EEPROM_SUPPORT
 #include "i2c_eeprom_api.h"
 #include "mw_eeprom_api.h"
diff --git a/src/main.c b/src/main.c
index 9f1dba9..f09cf5a 100644
--- a/src/main.c
+++ b/src/main.c
@@ -26,10 +26,9 @@
 
 #include "flashcmd_api.h"
 #include "ch341a_spi.h"
-#include "spi_nand_flash.h"
 
 struct flash_cmd prog;
-extern unsigned int bsize;
+unsigned int bsize;
 
 #ifdef EEPROM_SUPPORT
 #include "ch341a_i2c.h"
@@ -52,9 +51,7 @@ extern int org;
 void title(void)
 {
 #ifdef EEPROM_SUPPORT
-	printf("\nSNANDer - Serial Nor/nAND/Eeprom programmeR v." _VER " by McMCC <mcmcc@mail.ru>\n\n");
-#else
-	printf("\nSNANDer - Spi Nor/nAND programmER v." _VER " by McMCC <mcmcc@mail.ru>\n\n");
+	printf("\nSNANDer - Serial eeprom programmer v." _VER " by McMCC <mcmcc@mail.ru>\n\n");
 #endif
 }
 
@@ -63,8 +60,6 @@ void usage(void)
 	const char use[] =
 		"  Usage:\n"\
 		" -h             display this message\n"\
-		" -d             disable internal ECC(use read and write page size + OOB size)\n"\
-		" -I             ECC ignore errors(for read test only)\n"\
 		" -L             print list support chips\n"\
 		" -i             read the chip ID info\n"\
 		"" EHELP ""\
@@ -89,9 +84,7 @@ int main(int argc, char* argv[])
 	title();
 
 #ifdef EEPROM_SUPPORT
-	while ((c = getopt(argc, argv, "diIhveLl:a:w:r:E:f:8")) != -1)
-#else
-	while ((c = getopt(argc, argv, "diIhveLl:a:w:r:")) != -1)
+	while ((c = getopt(argc, argv, "ihveLl:a:w:r:E:f:8")) != -1)
 #endif
 	{
 		switch(c)
@@ -140,13 +133,6 @@ int main(int argc, char* argv[])
 				}
 				break;
 #endif
-			case 'I':
-				ECC_ignore = 1;
-				break;
-			case 'd':
-				ECC_fcheck = 0;
-				_ondie_ecc_flag = 0;
-				break;
 			case 'l':
 				str = strdup(optarg);
 				len = strtoll(str, NULL, *str && *(str + 1) == 'x' ? 16 : 10);
@@ -184,11 +170,6 @@ int main(int argc, char* argv[])
 
 	if (op == 0) usage();
 
-	if (op == 'x' || (ECC_ignore && !ECC_fcheck) || (op == 'w' && ECC_ignore)) {
-		printf("Conflicting options, only one option at a time.\n\n");
-		return -1;
-	}
-
 	if (ch341a_spi_init() < 0) {
 		printf("Programmer device not found!\n\n");
 		return -1;
@@ -202,8 +183,6 @@ int main(int argc, char* argv[])
 		printf("Programmer not supported auto detect EEPROM!\n\n");
 		goto out;
 	}
-#else
-	if (op == 'i') goto out;
 #endif
 
 	if (op == 'e') {
diff --git a/src/snander.rc b/src/snander.rc
index 939ceeb..496b288 100644
--- a/src/snander.rc
+++ b/src/snander.rc
@@ -12,7 +12,7 @@ BEGIN
   BEGIN
    BLOCK "040904E4"
    BEGIN
-    VALUE "FileDescription", "SNANDer - Serial Nor/nAND/Eeprom programmeR\0"
+    VALUE "FileDescription", "SNANDer - Serial eeprom programmer\0"
     VALUE "OriginalFilename", "SNANDer.exe\0"
     VALUE "CompanyName", "-= (c) 2018-2022 Igor Mokrushin aka McMCC =-\0"
     VALUE "FileVersion", "1.7.5.0\0"
